eureka/S5_lightcurve_fitting/plots_s5.py

```diff
@@ -28,27 +28,53 @@ def plot_fit(lc, model, meta, fitter):
     History:
     - December 29, 2021 Taylor Bell
         Moved plotting code to a separate function.
+    - January 7-22, 2022 Megan Mansfield
+        Adding ability to do a single shared fit across all channels
     """
     if type(fitter)!=str:
         raise ValueError('Expected type str for fitter, instead received a {}'.format(type(fitter)))
     
-    model_lc = model.eval()
-    residuals = (lc.flux - model_lc) #/ lc.unc
+    if lc.share:
```

Comment:
It might make more sense to leave this function as it was and put `if lc.share:' and the for loop around the call to `plot_fit()`. Does that make sense?

Reply:
The problem with that is that plot_fit just takes in the lightcurve object, not a time and a flux or something like that. So when you call lc.flux in plot_fit it gives you an array that's all of the channels combined. That's why I put this all in the plotting, because you need a way to separate out the different channels from the large lc.flux object. But perhaps I'm missing an easy way to fix this problem?

Reply:
Can we just change how this function is called to have it take in arrays rather than model and lc objects to avoid this code duplication?

Reply:
Oh wait, my same trick of setting lc.nchannel=1 will also work here to get rid of the if/else pair. The only thing you'll have to be careful about is that zfill function call which would be affected by the nchannel=1 change. Perhaps you can precompute a lc.zfill parameter inside the lc.__init__ function and then set lc.nchannel=1 after you've computed the zfill value?

Reply:
Done with the nchannel=1 thing. I didn't get any errors from the zfill stuff, but check it out and see how it looks.